Browser/CefSharpBrowser/CefOp/GadgetUrlHandler.cs

```diff
@@ -1,15 +1,30 @@
-﻿using CefSharp;
+﻿using BrowserLibCore;
+using CefSharp;
 using CefSharp.Handler;
+using IBrowser = CefSharp.IBrowser;
 
 namespace Browser.CefSharpBrowser.CefOp;
 
 internal class GadgetUrlHandler : ResourceRequestHandler
 {
+	private GadgetServerOptions GadgetBypassServer { get; }
+	private string GadgetBypassServerCustom {get; }
+	public GadgetUrlHandler(GadgetServerOptions gadgetBypassServer, string gadgetBypassServerCustom)
+	{
+		GadgetBypassServer = gadgetBypassServer;
+		GadgetBypassServerCustom = gadgetBypassServerCustom;
+	}
 	protected override IResponseFilter GetResourceResponseFilter(IWebBrowser chromiumWebBrowser, IBrowser browser, IFrame frame, IRequest request, IResponse response)
 	{
 		if (request.Url.Contains(@"gadget_html5"))
 		{
-			return new GadgetReplaceFilter("http://203.104.209.7/gadget_html5/", "https://kcwiki.github.io/cache/gadget_html5/");
+			return GadgetBypassServer switch
+			{
+				GadgetServerOptions.Wiki => new GadgetReplaceFilter("http://203.104.209.7/gadget_html5/", "https://kcwiki.github.io/cache/gadget_html5/"),
+				GadgetServerOptions.EO => new GadgetReplaceFilter("http://203.104.209.7/gadget_html5/", "https://electronicobserveren.github.io/cache/gadget_html5/"),
+				GadgetServerOptions.Custom => new GadgetReplaceFilter("http://203.104.209.7/gadget_html5/", GadgetBypassServerCustom + "/gadget_html5/"),
```

Comment:
Hmm, will having the custom option being just the base address be okay here? These things can be confusing since the end user won't know if the base address has to finish with a slash or not. It might be better to just let the user specify the whole address, so users can just copy what the built-in options provide.
A tooltip to clarify the format would work too, I guess.